dpe/src/commands/derive_child.rs

```diff
@@ -95,10 +92,20 @@ impl<C: Crypto, P: Platform> CommandExecution<C, P> for DeriveChildCmd {
         if !dpe.support.internal_info && self.uses_internal_info_input()
             || !dpe.support.internal_dice && self.uses_internal_dice_input()
         {
-            return Err(DpeErrorCode::InvalidArgument);
+            return Err(DpeErrorCode::ArgumentNotSupported);
+        }
+
+        if !dpe.support.is_ca && self.allows_ca() || !dpe.support.x509 && self.allows_x509() {
```

Comment:
I think the two clauses here need parentheses

```suggestion
        if (!dpe.support.is_ca && self.allows_ca()) || (!dpe.support.x509 && self.allows_x509()) {
```